library/ssl_tls.c

```diff
@@ -1118,12 +1167,12 @@ int mbedtls_ssl_derive_keys( mbedtls_ssl_context *ssl )
             else
                 alg = PSA_ALG_TLS12_PSK_TO_MS(PSA_ALG_SHA_256);
 
-            status = psa_key_derivation( &derivation, psk, alg,
-                                         salt, salt_len,
-                                         (unsigned char const *) lbl,
-                                         (size_t) strlen( lbl ),
-                                         master_secret_len );
-            if( status != PSA_SUCCESS )
+            if ( !setup_psa_key_derivation( &derivation,
```

Comment:
Could we have setup_psa_key_derivation return a psa_status_t instead? We don't really like functions returning 1 when they pass. We prefer a Unix-style where success is 0. In addition, using a psa_status_t has the advantage that we can bubble up the underlying error directly.

Reply:
I chose to go for a boolean result as I found it odd to return a PSA status from a non PSA code. If it was just the "bubbling up", I'd go for it immediately, but I didn't feel comfortable with returning the "PSA unsuported" error when the truth is it's "helper function unsupported". Nevertheless, I force-pushed a solution that you suggested.

Reply:
It's accurate that our implementation doesn't support other key derivation algorithms in this use case when the helper function doesn't support it.